nextest-runner/src/runner.rs

```diff
@@ -1082,6 +1082,9 @@ impl<'a> TestRunnerInner<'a> {
                     () = sleep, if !child_fds.is_done() => {
                         break true;
                     }
+                    _ = forward_receiver.recv() => {
```

Comment:
This is happening because the above branches get disabled but this one doesn't. The fix is to also add `!child_fds.is_done()` to this branch so the `else` is hit.